examples/api/test/material/segmented_button/segmented_button.0_test.dart

```diff
@@ -0,0 +1,111 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/segmented_button/segmented_button.0.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Segmented button can be used for with a single selection', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.SegmentedButtonApp(),
+    );
+
```

Comment:
Not sure it will be more readable, maybe use a helper to avoid repeating the verbose expect?

```suggestion

    void expectOneCalendarButton(example.Calendar period) {
      expect(
        find.byWidgetPredicate(
          (Widget widget) => widget is SegmentedButton<example.Calendar> && setEquals(widget.selected, <example.Calendar>{period}),
        ),
        findsOne,
      );
    }
    
```